kernel/init/init.c

```diff
@@ -0,0 +1,53 @@
+//! @file init.c
+//! @brief File containing kernel entrypoint and kernel init functions
+
+#include <init/stivale2.h> // For stivale2 protocol type definitions
+#include <lib/log.h>	   // For log_printf
+#include <sys/ports.h>
+
+//! @brief Stack used by kernel on bootstrap
+//! As dictated by stivale2 protocol, we should provide an address of the stack
+//! that we will use to the bootloader. As bootloader needs to know the stack
+//! before kernel even starts, we allocate stack statically as a global
+//! uninitialized array
+static uint8_t kernel_stack[4096];
```

Comment:
Maybe add more stack space. I personally tend to use 64K, to avoid even having to think about stack overflow.

Reply:
fixed